Anagram.java

```diff
@@ -2,48 +2,75 @@
 public class Anagram {
 	public static void main(String args[]) {
 		// Tests the isAnagram function.
-		System.out.println(isAnagram("silent","listen"));  // true
-		System.out.println(isAnagram("William Shakespeare","I am a weakish speller")); // true
-		System.out.println(isAnagram("Madam Curie","Radium came")); // true
-		System.out.println(isAnagram("Tom Marvolo Riddle","I am Lord Voldemort")); // true
-
+		System.out.println(preProcess("William Shakespeare"));
+		System.out.println(preProcess("I am a weakish speller"));
+		System.out.println(isAnagram("William Shakespeare","I am a weakish speller"));  // 
+		//System.out.println(isAnagram("William Shakespeare","I am a weakish speller")); // true
+		//System.out.println(isAnagram("Madam Curie","Radium came")); // true
+		//System.out.println(isAnagram("Tom Marvolo Riddle","I am Lord Voldemort")); // true
+		
 		// Tests the preProcess function.
-		System.out.println(preProcess("What? No way!!!"));
+		//System.out.println(preProcess("Hello World!"));
 		
 		// Tests the randomAnagram function.
-		System.out.println("silent and " + randomAnagram("silent") + " are anagrams.");
+		//System.out.println("silent and " + randomAnagram("silent") + " are anagrams.");
 		
 		// Performs a stress test of randomAnagram 
-		String str = "1234567";
-		Boolean pass = true;
-		//// 10 can be changed to much larger values, like 1000
-		for (int i = 0; i < 10; i++) {
-			String randomAnagram = randomAnagram(str);
-			System.out.println(randomAnagram);
-			pass = pass && isAnagram(str, randomAnagram);
-			if (!pass) break;
+		//String str = "1234567";
+		//Boolean pass = true;
+		// 10 can be changed to much larger values, like 1000
+		//for (int i = 0; i < 10; i++) {
+		//	String randomAnagram = randomAnagram(str);
+		//	System.out.println(randomAnagram);
+		//	pass = pass && isAnagram(str, randomAnagram);
+		//	if (!pass) break;
 		}
-		System.out.println(pass ? "test passed" : "test Failed");
-	}  
+		//System.out.println(pass ? "test passed" : "test Failed");
+	//}
+	
 
 	// Returns true if the two given strings are anagrams, false otherwise.
 	public static boolean isAnagram(String str1, String str2) {
-		// Replace the following statement with your code
-		return false;
-	}
+		String str11 = preProcess(str1);
+		String str21 = preProcess(str2);
+		boolean isAnagram = true; 
+			for(int i = 0; i < str11.length(); i++) {
+				for(int j = 0; j < str21.length(); j++) {
+					if(str11.indexOf(str21.charAt(j)) != -1)
+					isAnagram = isAnagram && true;
+					else 
+					isAnagram = isAnagram && false;
```

Comment:
OK lets discuss these 2 lines:   
1. `isAnagram = isAnagram && true;`
2. `isAnagram = isAnagram && false;`

lets start with 1, we agree that:
1. `true && true = true;` 
2. `false && true = false;`
We see that in both options the left one decides the expected value lets mark it `flag` , update and combine them into 1 ---> `flag && true = flag;`  
We can see it is resembles `x * 1 = x`,`x + 0 = x` . ,`string + "" = string` . 
We can deduct that `true` is neutral value for `&&`. 
I could have switch them up in order since `flag && flag1 == flag1 && flag`.
Therefore  `isAnagram = isAnagram && true;` means `isAnagram = isAnagram;` which is plain redundant..

lets move to 2, we agree that:
1. `true && false = false;` 
2. `false && false = false;`
We see that in both options the false is decides the expected value lets mark it no matter what is the other value, `flag`, update and combine them into 1 ---> `flag && false = false;`  
We can see it is resembles `x * 0 = 0`;
We can deduct that `false` has Zero-product property for `&&`. 
Therefore  `isAnagram = isAnagram && false;` means `isAnagram = false;`

now to switch in order to fix it
```
if(str11.indexOf(str21.charAt(j)) == -1){
    isAnagram = false;
    return false;
}
			
```